From 0ffd40396be7eac47014ce407c584e0a8149446d Mon Sep 17 00:00:00 2001
From: Santiago Seifert <aquilescanta@google.com>
Date: Tue, 16 Nov 2021 14:34:22 +0000
Subject: [PATCH] Prevent out-of-bounds read

Bug: 204077881
Test: See bug for PoC instructions.
Change-Id: Ie8d6792ff391d36d0e199a1d7f649e6081c1e8ac
Merged-In: Ie8d6792ff391d36d0e199a1d7f649e6081c1e8ac
(cherry picked from commit d55f55d2f8beffcfbdd3f2b66007ea6d4f851d73)
(cherry picked from commit cc538ed26803328445d52383f91025b357cda47d)
Merged-In:Ie8d6792ff391d36d0e199a1d7f649e6081c1e8ac
---
 media/libstagefright/rtsp/AAVCAssembler.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/media/libstagefright/rtsp/AAVCAssembler.cpp b/media/libstagefright/rtsp/AAVCAssembler.cpp
index 3f4d66249e..30cdbc93fd 100644
--- a/media/libstagefright/rtsp/AAVCAssembler.cpp
+++ b/media/libstagefright/rtsp/AAVCAssembler.cpp
@@ -295,6 +295,10 @@ ARTPAssembler::AssemblyStatus AAVCAssembler::addNALUnit(
 }
 
 void AAVCAssembler::checkSpsUpdated(const sp<ABuffer> &buffer) {
+    if (buffer->size() == 0) {
+        android_errorWriteLog(0x534e4554, "204077881");
+        return;
+    }
     const uint8_t *data = buffer->data();
     unsigned nalType = data[0] & 0x1f;
     if (nalType == 0x7) {
-- 
2.17.1

